compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/ProjectUtils.java

```diff
@@ -194,10 +196,18 @@ public static Set<String> getPackageImports(Package pkg) {
     private static void getPackageImports(Set<String> imports, Module module, Collection<DocumentId> documentIds) {
         for (DocumentId docId : documentIds) {
             Document document = module.document(docId);
-
             ModulePartNode modulePartNode = document.syntaxTree().rootNode();
-
             for (ImportDeclarationNode importDcl : modulePartNode.imports()) {
+                boolean isErrorInImport = false;
+                for (Diagnostic diagnostic : importDcl.diagnostics()) {
+                    if (diagnostic.diagnosticInfo().severity() == DiagnosticSeverity.ERROR) {
+                        isErrorInImport = true;
+                        break;
+                    }
+                }
+                if (isErrorInImport) {
+                    continue;
+                }
```

Comment:
```suggestion
            outerloop:
            for (ImportDeclarationNode importDcl : modulePartNode.imports()) {
                for (Diagnostic diagnostic : importDcl.diagnostics()) {
                    if (diagnostic.diagnosticInfo().severity() == DiagnosticSeverity.ERROR) {
                        continue outerloop;
                    }
                }
```
Would this be better?